File: tests/test_bloom.py
import unittest

from src import bloom


class BloomFilterTestCase(unittest.TestCase):
    def test_reset(self):
        bf = bloom.BloomFilter(100)
        self.assertEqual(bf._bits, [0 for i in range(100)])

    def test_add(self):
        bf = bloom.BloomFilter(100)
        bf.add("Hello")
        self.assertEqual(len([1 for bit in bf._bits if bit]), 5)

    def test_contains(self):
        bf = bloom.BloomFilter(100)
        bf.add("Hello")
        bf.add("world")
        bf.add("HELLO, YES, THIS IS DOG")

        self.assertTrue("Hello" in bf)
        self.assertTrue("world" in bf)
        self.assertTrue("HELLO, YES, THIS IS DOG" in bf)
        self.assertFalse("Welp" in bf)

File: tests/test_bloom.js
const test = require("tape");

const bloom = require("../src/bloom.js");

test("Testing bloom", function(t) {
  t.plan(6);

  let bf = new bloom.BloomFilter(100);
  t.deepEqual(bf.bits, Array(100).fill(0));

  bf.add("Hello");
  t.equal(bf.bits.filter(v => v === 1).length, 5);

  bf.add("world");
  bf.add("HELLO, YES, THIS IS DOG");
  t.equal(bf.test("Hello"), true);
  t.equal(bf.test("world"), true);
  t.equal(bf.test("HELLO, YES, THIS IS DOG"), true);
  t.equal(bf.test("Welp"), false);
});
